Automattic/jetpack
# Search: Modularize Customizer and DOM integration

#### Changes proposed in this Pull Request:
In preparation for using our Instant Search application within Gutenberg, this PR modularizes `<SearchApp>` to better isolate its Customizer and DOM integration.

#### Jetpack product discussion
Necessary for our future customization efforts, see p1HpG7-c6N-p2.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription.
* Ensure that the search interface works as expected.
